core/src/main/java/org/apache/calcite/sql/fun/SqlCastFunction.java

```diff
@@ -93,6 +95,29 @@ public SqlCastFunction() {
 
   //~ Methods ----------------------------------------------------------------
 
+  @Override public void validateCall(final SqlCall call, final SqlValidator validator,
+      final SqlValidatorScope scope,
+      final SqlValidatorScope operandScope) {
+    try {
+      super.validateCall(call, validator, scope, operandScope);
+    } catch (UnsupportedOperationException ex) {
+      assert call.operandCount() == 2;
```

Comment:
Could you please remove the assert from here?
What is the purpose of the assert?

Reply:
I forgot to remove the assert. This was added during development and test. 

I will remove it when I push the updates once the other comment is resolved.